sdk/dotnet/Pulumi.Automation/Commands/LocalPulumiCmd.cs

```diff
@@ -128,9 +128,9 @@ private static string SanitizeCommandName(string? firstArgument)
             return alphaNumWord.IsMatch(firstArgument) ? firstArgument : "event-log";
         }
 
-        private class EventLogFile : IDisposable
+        private sealed class EventLogFile : IDisposable
```

Comment:
This class was over complicated, using the [dispose pattern](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern), although without a destructor, when it didn't really need to. It's only used internally, and with a `using` statement, so I `sealed` it and stripped it back to a simple `Dispose`. I `sealed` other simple classes using `Dispose` as well.

Reply:
Ok no harm done here, simpler code is easier to read, and if it disposes twice it's idempotent so it's fine. Previous code probably was IDE-generated.